src/group_theory/Maschke.lean

```diff
@@ -0,0 +1,172 @@
+/-
```

Comment:
Do we use capitalized file names elsewhere? I'd be a little confused to have to `import group_theory.Maschke`.

Reply:
Sure, I'll rename. While we're at it, where does this file actually go?
`import algebra.rep_theory.maschke`? Or should we pre-emptively make `rep_theory` a top-level directory? :-)

Reply:
It shouldn't be a subdirectory of `algebra/` because that's only for the basic algebraic hierarchy. I'm fine with putting this in `group_theory/` but a new top-level `representation_theory/` also sounds good.

Reply:
Another option is to make `representation_theory` a subfolder of the `group_theory` folder.

Reply:
I think few representation theorists would say that their subject was a subdiscipline of group theory. Representation theory is about linear actions (whether of groups, algebras, or even higher categories) on additive things (whether modules, vector spaces, or even linear categories). Certainly representations of finite groups is a lovely initial part of the subject, but it's thoroughly intertwined (pun intended?) with representations of everything else.

Reply:
ok, then I would make it a top level directory. I would write `representation_theory` out fully, though.

Reply:
Okay, I've removed capitalisation, and moved this to `representation_theory/maschke.lean`.